tests/Silverback.Integration.Tests.E2E/Kafka/MessageValidationTests.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Silverback.Messaging.Configuration;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Publishing;
+using Silverback.Messaging.Validation;
+using Silverback.Tests.Integration.E2E.TestHost;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Silverback.Tests.Integration.E2E.Kafka
+{
+    public class MessageValidationTests : KafkaTestFixture
+    {
+        public MessageValidationTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        public static IEnumerable<object[]> InvalidMessages()
+        {
+            yield return new object[]
+            {
+                new Message { Id = "1", String10 = "123456789abc", IntRange = 5, NumbersOnly = "123" },
+                "The field String10 must be a string with a maximum length of 10."
+            };
+            yield return new object[]
+            {
+                new Message { Id = "1", String10 = "123456", IntRange = 30, NumbersOnly = "123" },
+                "The field IntRange must be between 5 and 10."
+            };
+            yield return new object[]
+            {
+                new Message { String10 = "123456", IntRange = 5, NumbersOnly = "123" },
+                "The Id field is required."
+            };
+            yield return new object[]
+            {
+                new Message { Id = "1", String10 = "123456", IntRange = 5, NumbersOnly = "Test1234" },
+                "The field NumbersOnly must match the regular expression '^[0-9]*$'."
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidMessages))]
+        public async Task Validation_WithException_InvalidMessageNotProduced(Message message, string expectedValidationMessage)
+        {
+            // Arrange
```

Comment:
We don't do the `// Arrange // Act // Assert` thing here.

Also get rid of the commented out code below.